bundles/async_remote_services/admin/src/admin.cc

```diff
@@ -178,143 +171,121 @@ void celix::async_rsa::AsyncAdmin::removeExportedServiceFactory(celix::async_rsa
+            _toBeExportedServices.erase(it);
+            break;
+        }
     }
 
-    auto endpointId = properties.get(ENDPOINT_IDENTIFIER);
-
-    if(endpointId.empty()) {
-        L_WARN("Adding endpoint but missing service id");
-        return;
-    }
+}
 
-    auto existingFactory = _importedServiceFactories.find(interface);
+void celix::async_rsa::AsyncAdmin::createExportServices() {
+    std::lock_guard<std::mutex> lock{_m};
 
-    if(existingFactory == end(_importedServiceFactories)) {
-        L_DEBUG("Adding endpoint but no factory available yet, delaying creation");
-        _toBeCreatedImportedEndpoints.emplace_back(celix::rsa::Endpoint{properties});
-        return;
+    for (auto it = _toBeExportedServices.begin(); it != _toBeExportedServices.end(); ++it) {
+        auto serviceName = it->second->get(celix::SERVICE_NAME, "");
+        auto svcId = it->second->getAsLong(celix::SERVICE_ID, -1);
+        if (serviceName.empty()) {
+            L_WARN("Adding service to be exported but missing objectclass for svc id %li", svcId);
+        } else {
+            auto factory = _exportedServiceFactories.find(serviceName);
+            if (factory != _exportedServiceFactories.end()) {
+                auto endpointId = _ctx->getFramework()->getUUID() + "-" + std::to_string(svcId);
+                auto cmpUUID = factory->second->create(it->first.get(), std::move(endpointId)).getUUID();
+                _exportedServiceInstances.emplace(svcId, std::move(cmpUUID));
+                _toBeExportedServices.erase(it--);
+            } else {
+                L_DEBUG("Adding service to be exported but no factory available yet, delaying creation");
+            }
+        }
     }
+}
 
-    L_DEBUG("Adding endpoint, created service");
-    _importedServiceInstances.emplace(endpointId, existingFactory->second->create(endpointId));
+void celix::async_rsa::AsyncAdmin::setLogger(const std::shared_ptr<celix_log_service>& logService) {
+    _logService = logService;
 }
 
 class AdminActivator {
 public:
-    explicit AdminActivator(std::shared_ptr<celix::dm::DependencyManager> mng) :
-            _cmp(mng->createComponent(std::make_unique<celix::async_rsa::AsyncAdmin>(mng))), _mng(std::move(mng)) {
+    explicit AdminActivator(const std::shared_ptr<celix::BundleContext>& ctx) {
+        auto admin = std::make_shared<celix::async_rsa::AsyncAdmin>(ctx);
 
-        _cmp.createServiceDependency<celix::rsa::Endpoint>()
+        auto& cmp = ctx->getDependencyManager()->createComponent(admin);
+        cmp.createServiceDependency<celix::rsa::Endpoint>()
                 .setRequired(false)
-                .setCallbacks(&celix::async_rsa::AsyncAdmin::addEndpoint, &celix::async_rsa::AsyncAdmin::removeEndpoint)
-                .build();
-
-        _cmp.createServiceDependency<celix::async_rsa::IImportedServiceFactory>()
+                .setStrategy(celix::dm::DependencyUpdateStrategy::locking)
```

Comment:
I see that this is missing some documentation on the enum values.

For service dependency the suspend strategy will stop and start the component for every update (note that this is not really relevant for the admin, because it has no stop/start). 
The locking strategy will use the dependency is protected with a lock, but has no locking on it self.